experiment/scheduler.py

```diff
@@ -106,8 +106,12 @@ def all_trials_ended(experiment: str) -> bool:
 
 def delete_instances(instances, experiment_config):
     """Deletes |instances|."""
-    running_instances = gcloud.list_instances()
-    instances_to_delete = [i for i in instances if i in running_instances]
+    cloud_project = experiment_config['cloud_project']
```

Comment:
I'm not sure this works 100% properly.
I think the old code also got terminated instances (aka instances that aren't running but were stopped), despite the variable running_instances implying otherwise.
I'm not sure the bug here is a big deal though, worst case scenario we might not delete some preempted instances (in which case we might have to pay for their disk space).

Reply:
Actually nevermind, this calls into an equivalent function.